pkg/registry/ociregistry.go

```diff
@@ -0,0 +1,164 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// OCIRegistryClient storage client for an OCI registry.
+type OCIRegistryClient struct {
+	Host        string
+	Project     string
+	CertFile    string
+	Insecure    bool
+	DryRun      bool
```

Comment:
do these need to be exported if they can be set through the constructor?

Reply:
Made them private